src/extension/kernel.ts

```diff
@@ -734,6 +738,70 @@ export class Kernel implements Disposable {
     })
   }
 
+  async #handleTrackCellRun(
+    cell: NotebookCell,
+    successfulCellExecution: boolean,
+    startTime: number,
+    endTime: number,
+  ): Promise<void> {
+    const session = await getPlatformAuthSession(false)
+
+    if (!this.serializer?.isLifecycleIdentity(RunmeIdentity.ALL || !session)) {
```

Comment:
I think the `||` should after closing the call `isLifecycleIdentity`.

```suggestion
    if (!this.serializer?.isLifecycleIdentity(RunmeIdentity.ALL) || !session) {
```

Reply:
Apart of that, I think is not necessary to add a new method to the serializer. It's enough calling the runtime setting value directly from `ContextState.getKey<ServerLifecycleIdentity>(NOTEBOOK_LIFECYCLE_ID)`.

Reply:
Ha! Totally. Last minute change to make readability better and I missed that nonsense parenthesis. Good catch!

Reply:
Fixed in 99e68515e54b2cb2745ea9658518dfc16b5cd32e